cli/internal/run/real_run.go

```diff
@@ -90,12 +91,14 @@ func RealRun(
 		Concurrency: rs.Opts.runOpts.concurrency,
 	}
 
+	var logMutex sync.Mutex
```

Comment:
I don't think this mutex is in the right place. If we need it at the level of task execution, let's attach it to something on the `execContext`. However, I suspect we are better off localizing it to the log replayer implementation

Reply:
The mutex has to be usable across package tasks, however this will still be removed when switching to the queue based implementation.